src/main/java/uk/gov/justice/digital/hmpps/keyworker/config/WebClientConfiguration.kt

```diff
@@ -44,25 +44,27 @@ class WebClientConfiguration(
   }
 
   @Bean
-  fun webClient(builder: WebClient.Builder): WebClient {
-    return builder
-      .baseUrl(elite2ApiRootUri)
-      .filter(addAuthHeaderFilterFunction())
+  fun complexityOfNeedWebClient(authorizedClientManager: OAuth2AuthorizedClientManager?, builder: WebClient.Builder): WebClient {
+    val oauth2Client = ServletOAuth2AuthorizedClientExchangeFilterFunction(authorizedClientManager)
+    oauth2Client.setDefaultClientRegistrationId("elite2-api")
```

Comment:
i assume this is for convenience - cos it should ideally be keyworker-api as the client

Reply:
Yeah, it's confusing. Once decoded the client id is is in fact keyworker-api. I just need to rename all the config so it aligns.